### PR TITLE
Fix setTag logic to prevent high cpu load on rds

### PR DESCRIPTION
**Pull request by stevie9868**
For now, whenever we want to set a tag on a resource, it will always fetch all tags from the dbs into memory, and then do a diff to figure out what tags never exist in the db. Loading all tags at once creates a lot of load on the rds if the number of tags is huge when number of setTag request is high. 

For this PR, I also remove some unused code.

Instead of fetching all tags into memory, we will instead do 
```
"INSERT INTO lookup_values(lookup_id, values_string) VALUES (?,?) "
            + "ON DUPLICATE KEY UPDATE lookup_id=lookup_id, values_string=values_string"; 
```

Even though, there is no unit test for tag component, there are lots of existing tests in the functional test, and I also add one more functional test to cover more scenarios.
Since the logic I change should have no impact on the core logic, we should expect the functionalTest to have the same result [before](https://github.com/Netflix/metacat/pull/573) and after this change.

Finally, in order to make this insert query fast, we will need to build a composite index on (lookup_id, values_string)

### CODE REVIEW

File: metacat-metadata-mysql/src/test/groovy/com/netflix/metacat/metadata/mysql/MySqlLookupServiceSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    def setup() {
+        config = Mock(Config)
+        jdbcTemplate = Mock(JdbcTemplate)
+
+        mySqlLookupService = new MySqlLookupService(config, jdbcTemplate)
+    }
+
+    @Unroll
+    def "test includeValues"() {
```
Comment (insyncoss) [inline metacat-metadata-mysql/src/test/groovy/com/netflix/metacat/metadata/mysql/MySqlLookupServiceSpec.groovy:48]: Can you add some additional tests featuring duplicates and multiple call iterations

Comment (stevie9868) [reply metacat-metadata-mysql/src/test/groovy/com/netflix/metacat/metadata/mysql/MySqlLookupServiceSpec.groovy:48]: Sorry, could you elaborate more?

I think I have covered them also in the functional test.

Comment (insyncoss) [reply metacat-metadata-mysql/src/test/groovy/com/netflix/metacat/metadata/mysql/MySqlLookupServiceSpec.groovy:48]: ok looks like the functional test can cover the iterations part, could you add another unit test featuring if a caller tries to add duplicate tag in one call like pass in the same tag multiple times in the input list

File: metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlLookupService.java [hunk 2]
```diff
@@ -244,20 +206,14 @@ private Lookup findOrCreateLookupByName(final String name) throws SQLException {
...
-                lookup.setValues(values);
-            } else {
-                inserts = Sets.difference(values, lookupValues);
+            final Lookup lookup = findOrCreateLookupByName(name, includeValues);
+            if (!values.isEmpty()) {
+                insertLookupValuesIfNotExist(lookup.getId(), values);
             }
-            if (!inserts.isEmpty()) {
-                insertLookupValues(lookup.getId(), inserts);
+            if (includeValues) {
```
Comment (insyncoss) [inline metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlLookupService.java:215]: the previous code didn't even add back the new tags to the return result, but based on usage it does not make a difference either way

Comment (stevie9868) [reply metacat-metadata-mysql/src/main/java/com/netflix/metacat/metadata/mysql/MySqlLookupService.java:215]: yep, you are right but I want to make it consistent.

**Review by insyncoss — APPROVED** [review]
